### PR TITLE
Add MIMXRT685 FlexSPI XIP support.

### PR DESCRIPTION
**Pull request by saurabh-nxp**
Driver APIs are tested using sample applications:
- samples/drivers/flash_shell 
- samples/subsys/fs/littlefs

These changes have dependency on PR #34086. So that MPU is setup to allow execution of driver code relocated to SRAM.

### CODE REVIEW

File: boards/arm/mimxrt685_evk/flash_flexspi_nor_lut.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "flash_mcux_flexspi_nor_lut_index.h"
+
+static const uint32_t flash_flexspi_nor_lut[][4] = {
```
Comment (MaureenHelm) [inline boards/arm/mimxrt685_evk/flash_flexspi_nor_lut.h:9]: Can this LUT be used for other octal flashes or is it specific to MX25UM51345G?

File: boards/arm/mimxrt685_evk/flash_flexspi_nor_lut.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+				kFLEXSPI_Command_READ_SDR,  kFLEXSPI_1PAD, 0x04),
+	},
+
+	[READ_STATUS_REG] = {
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_SDR, kFLEXSPI_8PAD, 0x05,
+				kFLEXSPI_Command_SDR, kFLEXSPI_8PAD, 0xFA),
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_RADDR_SDR, kFLEXSPI_8PAD, 0x20,
+				kFLEXSPI_Command_DUMMY_SDR, kFLEXSPI_8PAD, 0x14),
+		FLEXSPI_LUT_SEQ(kFLEXSPI_Command_READ_SDR, kFLEXSPI_8PAD, 0x04,
+				kFLEXSPI_Command_STOP, kFLEXSPI_1PAD, 0x0),
```
Comment (MaureenHelm) [inline boards/arm/mimxrt685_evk/flash_flexspi_nor_lut.h:21]: Please fix the whitespace to follow the convention in the existing LUT

File: boards/arm/mimxrt1064_evk/flash_flexspi_nor_lut.h [hunk 3]
```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2021 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "flash_mcux_flexspi_nor_lut_index.h"
+
+static const uint32_t flash_flexspi_nor_lut[][4] = {
```
Comment (MaureenHelm) [inline boards/arm/mimxrt1064_evk/flash_flexspi_nor_lut.h:9]: I don't like defining the LUT at the board level because it will have to be duplicated for every other board that uses the same flash. I'd prefer instead to keep the LUT defined in `drivers/flash` because that allows reuse by both in-tree and out-of-tree boards. I think we should either 1) conditionalize the LUT in `drivers/flash/flash_mcux_flexspi_nor.c`, or 2) introduce a new `drivers/flash/flash_mcux_flexspi_ospi.c` or `drivers/flash/flash_mcux_flexspi_mx25um51345g.c`. I'm leaning toward option 2.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: boards/arm/mimxrt685_evk/flash_flexspi_nor_lut.h (hunk 1 above)
Comment (saurabh-nxp) [reply boards/arm/mimxrt685_evk/flash_flexspi_nor_lut.h:9]: It is specific to MX25UM51345G. JEDEC standard is not valid for octal mode.

File: boards/arm/mimxrt1064_evk/flash_flexspi_nor_lut.h (hunk 3 above)
Comment (saurabh-nxp) [reply boards/arm/mimxrt1064_evk/flash_flexspi_nor_lut.h:9]: So `drivers/flash/flash_mcux_flexspi_mx25um51345g.c` would be required since LUT is memory specific. If not duplicating LUT header, is duplicating source code `flash_mcux_<memoryName>.c`  acceptable for planned device support in future?

File: boards/arm/mimxrt685_evk/pinmux.c [hunk 4]
```diff
@@ -278,6 +278,239 @@ static int mimxrt685_evk_pinmux_init(const struct device *dev)
...
+		IOPCTL_PIO_FULLDRIVE_EN |
+		/* Analog mux is disabled */
+		IOPCTL_PIO_ANAMUX_DI |
+		/* Pseudo Output Drain is disabled */
+		IOPCTL_PIO_PSEDRAIN_DI |
+		/* Input function is not inverted */
+		IOPCTL_PIO_INV_DI);
+/* PORT2 PIN23 (coords: P5) is configured as FLEXSPI0B_DATA7 */
+IOPCTL_PinMuxSet(IOPCTL, 2U, 23U, port2_pin23_config);
+#endif
```
Comment (pdgendt) [inline boards/arm/mimxrt685_evk/pinmux.c:512]: It kind of hurts my eyes, to see the same property and comment so many times, it is consistent with the code above though.

Maybe simplify to (untested & copy-pasted)
```c
const uint32_t common_pin_config = (
		/* Disable pull-up / pull-down function */
		IOPCTL_PIO_PUPD_DI |
		/* Enable pull-down function */
		IOPCTL_PIO_PULLDOWN_EN |
		/* Enables input buffer function */
		IOPCTL_PIO_INBUF_EN |
		/* Normal mode */
		IOPCTL_PIO_SLEW_RATE_NORMAL |
		/* Full drive enable */
		IOPCTL_PIO_FULLDRIVE_EN |
		/* Analog mux is disabled */
		IOPCTL_PIO_ANAMUX_DI |
		/* Pseudo Output Drain is disabled */
		IOPCTL_PIO_PSEDRAIN_DI |
		/* Input function is not inverted */
		IOPCTL_PIO_INV_DI);

/* PORT1 PIN11 (coords: L2) is configured as FLEXSPI0B_DATA0 */
IOPCTL_PinMuxSet(IOPCTL, 1U, 11U, common_pin_config | IOPCTL_PIO_FUNC6);
/* PORT1 PIN12 (coords: M2) is configured as FLEXSPI0B_DATA1 */
IOPCTL_PinMuxSet(IOPCTL, 1U, 12U, common_pin_config | IOPCTL_PIO_FUNC6);

/* etc */
```

But ignore me, if this is a policy :)

File: drivers/memc/memc_mcux_flexspi.c [hunk 5]
```diff
@@ -140,6 +140,8 @@ static int memc_flexspi_init(const struct device *dev)
 #define MEMC_FLEXSPI_CFG_XIP(node_id) DT_SAME_NODE(node_id, DT_NODELABEL(flexspi))
 #elif defined(CONFIG_XIP) && defined(CONFIG_CODE_FLEXSPI2)
 #define MEMC_FLEXSPI_CFG_XIP(node_id) DT_SAME_NODE(node_id, DT_NODELABEL(flexspi2))
+#elif defined(CONFIG_SOC_SERIES_IMX_RT6XX)
+#define MEMC_FLEXSPI_CFG_XIP(node_id) true
```
Comment (pdgendt) [inline drivers/memc/memc_mcux_flexspi.c:144]: It solely depends on the soc type here, but XIP is enabled in the board defconfig.

```suggestion
#define MEMC_FLEXSPI_CFG_XIP(node_id) IS_ENABLED(CONFIG_FLASH_MCUX_FLEXSPI_XIP)
```
Or
```suggestion
#define MEMC_FLEXSPI_CFG_XIP(node_id) IS_ENABLED(CONFIG_XIP)
```

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts [hunk 6]
```diff
@@ -143,6 +143,19 @@ arduino_spi: &flexcomm5 {
...
+	mx25um51345g: mx25um51345g@0 {
+		compatible = "nxp,imx-flexspi-nor";
+		size = <536870912>;
+		label = "MX25UM51345G";
+		reg = <2>;
+		spi-max-frequency = <200000000>;
+		status = "okay";
+		jedec-id = [c2 81 3a];
+	};
+};
```
Comment (pdgendt) [inline boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts:157]: Move the relevant `reg` part to the flash device properties instead of the flash controller.

Similar to [PR](https://github.com/zephyrproject-rtos/zephyr/pull/34428)

File: boards/arm/mimxrt685_evk/flash_flexspi_nor_lut.h (hunk 2 above)
Comment (saurabh-nxp) [reply boards/arm/mimxrt685_evk/flash_flexspi_nor_lut.h]: Done.

File: boards/arm/mimxrt685_evk/flash_flexspi_nor_lut.h (hunk 1 above)
Comment (saurabh-nxp) [reply boards/arm/mimxrt685_evk/flash_flexspi_nor_lut.h:9]: Not just memory specific but to do XIP or to trigger a read command sequence from LUT, dummy cycles are required. And dummy cycles requires as per memory model's data sheet varies as per frequency at which flash is operated. So along with memory model LUT is also platform/board specific.

File: boards/arm/mimxrt685_evk/pinmux.c (hunk 4 above)
Comment (saurabh-nxp) [reply boards/arm/mimxrt685_evk/pinmux.c:512]: I just tried to keep it consistent with NXP's SDK code.

Comment (MaureenHelm) [reply boards/arm/mimxrt685_evk/pinmux.c:512]: I agree it's a bit overkill, but it is consistent with the rest of the file. This will soon go away anyway when we move the i.MX RT family to device tree pinmuxing.

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts (hunk 6 above)
Comment (MaureenHelm) [reply boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts:157]: Let's get agreement on the direction in that PR first, before we change this one to follow.

File: boards/arm/mimxrt1064_evk/flash_flexspi_nor_lut.h (hunk 3 above)
Comment (MaureenHelm) [reply boards/arm/mimxrt1064_evk/flash_flexspi_nor_lut.h:9]: Yes

Comment (saurabh-nxp) [reply boards/arm/mimxrt1064_evk/flash_flexspi_nor_lut.h:9]: Ok, will do. Please take  a note that devices to be supported in future with same memory model may not be able to use this LUT. Because LUT, specifically dummy cycles used for READ sequence would change based on frequency at which flash is operated. Also, this LUT is valid only for Octal mode but in case of SPI/DPI/QPI modes with same memory model would require an update to LUT.

File: boards/arm/mimxrt685_evk/Kconfig.defconfig [hunk 7]
```diff
@@ -14,6 +14,15 @@ config XTAL_SYS_CLK_HZ
...
 	default 260
 
+config FLASH_MCUX_FLEXSPI_NOR
+	default y if FLASH
+
+config FLASH_MCUX_FLEXSPI_XIP
+	default y
+
+config FLASH_MCUX_FLEXSPI_XIP_MEM
+	default "SRAM"
```
Comment (mmahadevan108) [inline boards/arm/mimxrt685_evk/Kconfig.defconfig:24]: Use `config FLASH_MCUX_FLEXSPI_XIP_MEM_TARGET` instead of `config FLASH_MCUX_FLEXSPI_XIP_MEM`. Else `#define CONFIG_FLASH_MCUX_FLEXSPI_XIP_MEM_ITCM 1` is set

File: boards/arm/mimxrt1064_evk/flash_flexspi_nor_lut.h (hunk 3 above)
Comment (saurabh-nxp) [reply boards/arm/mimxrt1064_evk/flash_flexspi_nor_lut.h]: Done, moved LUT to driver.

File: drivers/memc/memc_mcux_flexspi.c (hunk 5 above)
Comment (saurabh-nxp) [reply drivers/memc/memc_mcux_flexspi.c]: Done. Updated to `#define MEMC_FLEXSPI_CFG_XIP(node_id) IS_ENABLED(CONFIG_XIP)`

File: boards/arm/mimxrt685_evk/Kconfig.defconfig (hunk 7 above)
Comment (saurabh-nxp) [reply boards/arm/mimxrt685_evk/Kconfig.defconfig]: Done.

**Comment by mmahadevan108** [comment]
Please update the https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/mimxrt685_evk/doc/index.rst#supported-features and https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/mimxrt685_evk/doc/index.rst#connections-and-ios sections of the document

**Comment by mmahadevan108** [comment]
Can you take a look at PR https://github.com/zephyrproject-rtos/zephyr/pull/34789 and add a partition section to the dts

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts [hunk 8]
```diff
@@ -176,6 +176,18 @@ i2s1: &flexcomm3 {
...
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		storage_partition: partition@3f00000 {
+			label = "storage";
+			reg = <0x03f00000 DT_SIZE_M(1)>;
+		};
+	};
```
Comment (mmahadevan108) [inline boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts:190]: This commit can be squashed with the earlier one that changes board dts file

File: drivers/flash/CMakeLists.txt [hunk 9]
```diff
@@ -18,11 +18,19 @@ zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_NIOS2_QSPI soc_flash_nios2_qspi.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_GECKO flash_gecko.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_RV32M1 soc_flash_rv32m1.c)
 zephyr_library_sources_ifdef(CONFIG_FLASH_STM32_QSPI flash_stm32_qspi.c)
-zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_NOR flash_mcux_flexspi_nor.c)
+if(CONFIG_SOC_SERIES_IMX_RT6XX)
+	zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_NOR flash_mcux_flexspi_mx25um51345g.c)
+else()
+	zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_NOR flash_mcux_flexspi_nor.c)
+endif()
```
Comment (MaureenHelm) [inline drivers/flash/CMakeLists.txt:25]: Please add a new config to `Kconfig.mcux` to make this:
```
zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_MX25UM51345G flash_mcux_flexspi_mx25um51345g.c)
```

And have the board enable this new config instead of CONFIG_FLASH_MCUX_FLEXSPI_NOR

That will allow you to remove the SoC-specific conditional here.

File: drivers/flash/CMakeLists.txt [hunk 10]
```diff
@@ -50,7 +58,7 @@ zephyr_include_directories_ifdef(
 	CONFIG_SOC_FLASH_NRF_RADIO_SYNC_TICKER
 	${ZEPHYR_BASE}/subsys/bluetooth
 	${ZEPHYR_BASE}/subsys/bluetooth/controller/ll_sw/nordic
-	)
+)
```
Comment (MaureenHelm) [inline drivers/flash/CMakeLists.txt:61]: please revert this unrelated change

File: drivers/flash/flash_mcux_flexspi_mx25um51345g.c [hunk 11]
```diff
@@ -0,0 +1,542 @@
...
+		if (ret) {
+			LOG_ERR("Could not read status");
+			return ret;
+		}
+	} while (status & BIT(0));
+
+	return 0;
+}
+
+static int flash_flexspi_nor_enable_quad_mode(const struct device *dev)
```
Comment (MaureenHelm) [inline drivers/flash/flash_mcux_flexspi_mx25um51345g.c:284]: replace `quad` with `octal`?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/flash/CMakeLists.txt (hunk 9 above)
Comment (saurabh-nxp) [reply drivers/flash/CMakeLists.txt:25]: If I do that, I would have to add another config. based driver files inclusion in same CMakeLists.txt. Let me know if something as follows is ok to be added. Because this would duplicate existing `CONFIG_FLASH_MCUX_FLEXSPI_NOR` based option in some way.
`zephyr_include_directories_ifdef(
  CONFIG_FLASH_MCUX_FLEXSPI_MX25UM51345G
  ${ZEPHYR_BASE}/drivers/memc
)`

File: drivers/flash/CMakeLists.txt (hunk 10 above)
Comment (saurabh-nxp) [reply drivers/flash/CMakeLists.txt]: Done.

**Review by mmahadevan108 — APPROVED** [review]

File: drivers/flash/flash_mcux_flexspi_mx25um51345g.c (hunk 11 above)
Comment (saurabh-nxp) [reply drivers/flash/flash_mcux_flexspi_mx25um51345g.c]: Done.

File: drivers/flash/CMakeLists.txt (hunk 9 above)
Comment (saurabh-nxp) [reply drivers/flash/CMakeLists.txt]: Changes are pushed. Please check.

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts (hunk 8 above)
Comment (saurabh-nxp) [reply boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts:190]: After adding storage_partition to dts still facing an issue to get littlefs running. If able to get it to work will squash commits.

File: drivers/flash/CMakeLists.txt [hunk 12]
```diff
@@ -18,11 +18,16 @@ zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_NIOS2_QSPI soc_flash_nios2_qspi.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_GECKO flash_gecko.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_RV32M1 soc_flash_rv32m1.c)
 zephyr_library_sources_ifdef(CONFIG_FLASH_STM32_QSPI flash_stm32_qspi.c)
+zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_MX25UM51345G flash_mcux_flexspi_mx25um51345g.c)
 zephyr_library_sources_ifdef(CONFIG_FLASH_MCUX_FLEXSPI_NOR flash_mcux_flexspi_nor.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_FLASH_ESP32 flash_esp32.c)
 
 if(CONFIG_FLASH_MCUX_FLEXSPI_XIP)
-  zephyr_code_relocate(flash_mcux_flexspi_nor.c ${CONFIG_FLASH_MCUX_FLEXSPI_XIP_MEM}_TEXT)
+  if(CONFIG_SOC_SERIES_IMX_RT6XX)
```
Comment (MaureenHelm) [inline drivers/flash/CMakeLists.txt:26]: Replace `CONFIG_SOC_SERIES_IMX_RT6XX` with `CONFIG_FLASH_MCUX_FLEXSPI_MX25UM51345G`

File: drivers/flash/Kconfig.mcux [hunk 13]
```diff
@@ -34,10 +34,17 @@ config FLASH_MCUX_FLEXSPI_NOR
 	select MEMC
 	select MEMC_MCUX_FLEXSPI
 
+config FLASH_MCUX_FLEXSPI_MX25UM51345G
+	bool "MCUX FlexSPI NOR driver"
```
Comment (MaureenHelm) [inline drivers/flash/Kconfig.mcux:38]: Update this string to "MCUX FlexSPI MX25UM51345G driver"

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts [hunk 14]
```diff
@@ -165,6 +165,19 @@ i2s1: &flexcomm3 {
 	dma-names = "tx";
 };
 
+&flexspi {
+	reg = <0x50134000 0x4000>, <0x18000000 DT_SIZE_M(64)>;
+	mx25um51345g: mx25um51345g@0 {
+		compatible = "nxp,imx-flexspi-nor";
```
Comment (MaureenHelm) [inline boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts:171]: the compatible should be "nxp,imx-flexspi-mx25um51345g"

File: drivers/flash/flash_mcux_flexspi_mx25um51345g.c [hunk 15]
```diff
@@ -0,0 +1,550 @@
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT	nxp_imx_flexspi_nor
```
Comment (MaureenHelm) [inline drivers/flash/flash_mcux_flexspi_mx25um51345g.c:7]: this will become `nxp_imx_flexspi_mx25um51345g` after updating the dts compatible

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/flash/CMakeLists.txt (hunk 12 above)
Comment (saurabh-nxp) [reply drivers/flash/CMakeLists.txt]: Done.

File: drivers/flash/flash_mcux_flexspi_mx25um51345g.c (hunk 15 above)
Comment (saurabh-nxp) [reply drivers/flash/flash_mcux_flexspi_mx25um51345g.c]: Done.

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts (hunk 14 above)
Comment (saurabh-nxp) [reply boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts]: Done.

File: drivers/flash/Kconfig.mcux (hunk 13 above)
Comment (saurabh-nxp) [reply drivers/flash/Kconfig.mcux]: Done.

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts (hunk 8 above)
Comment (MaureenHelm) [reply boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts:190]: The sample is failing because flash_area is trying to use the flexspi device with the flash API, when it should use the mx25um51345g device instead. You can fix this by moving the partitions section inside the mx25um51345g node.

``` Booting Zephyr OS build zephyr-v2.5.0-3120-g39d7242197e9  ***
Area 0 at 0x3f00000 on MX25UM51345G for 1048576 bytes
/lfs mount: 0
/lfs: bsize = 16 ; frsize = 4096 ; blocks = 256 ; bfree = 254
/lfs/boot_count stat: 0
        fn 'boot_count' siz 4
/lfs/boot_count read count 1: 4
/lfs/boot_count seek start: 0
/lfs/boot_count write new boot count 2: 4
/lfs/boot_count close: 0
/lfs opendir: 0
  F 4 boot_count
End of files
/lfs unmount: 0
[00:16:13.131,023] <inf> littlefs: LittleFS version 2.2, disk version 2.0
[00:16:13.133,602] <inf> littlefs: FS at MX25UM51345G:0x3f00000 is 256 0x1000-byte blocks with 512 e
[00:16:13.133,605] <inf> littlefs: sizes: rd 16 ; pr 16 ; ca 64 ; la 32
[00:16:13.133,876] <inf> littlefs: /lfs mounted
[00:16:13.161,600] <inf> littlefs: /lfs unmounted
```

Comment (saurabh-nxp) [reply boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts:194]: Thank you! Done.

**Review by MaureenHelm — APPROVED** [review]

**Comment by saurabh-nxp** [comment]
I noticed some documentation and code compliance violation in CI workflows. Latest force-push addresses those only.

**Review by dleach02 — APPROVED** [review]

File: boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts [hunk 16]
```diff
@@ -170,6 +170,30 @@ i2s1: &flexcomm3 {
 	dma-names = "tx";
 };
 
+&flexspi {
+	reg = <0x50134000 0x4000>, <0x18000000 DT_SIZE_M(64)>;
+	mx25um51345g: mx25um51345g@0 {
```
[the remark this answers is not in the document]
Comment (saurabh-nxp) [reply boards/arm/mimxrt685_evk/mimxrt685_evk_cm33.dts]: Done.

File: drivers/flash/flash_mcux_flexspi_mx25um51345g.c [hunk 17]
```diff
@@ -0,0 +1,550 @@
+/*
+ * Copyright 2020 NXP
```
Comment (pdgendt) [inline drivers/flash/flash_mcux_flexspi_mx25um51345g.c:2]: Nit, we are 2021

File: drivers/flash/flash_mcux_flexspi_mx25um51345g.c [hunk 18]
```diff
@@ -0,0 +1,550 @@
...
+	transfer.data = NULL;
+	transfer.dataSize = 0;
+
+	LOG_DBG("Enabling write");
+
+	return memc_flexspi_transfer(data->controller, &transfer);
+}
+
+static int flash_flexspi_nor_erase_sector(const struct device *dev,
+	off_t offset)
```
Comment (pdgendt) [inline drivers/flash/flash_mcux_flexspi_mx25um51345g.c:212]: Nit, inconsistent indentation.

**Review by pdgendt — APPROVED** [review]

File: dts/bindings/mtd/nxp,imx-flexspi-mx25um51345g.yaml [hunk 19]
```diff
@@ -0,0 +1,8 @@
+# Copyright 2020 NXP
```
Comment (pdgendt) [inline dts/bindings/mtd/nxp,imx-flexspi-mx25um51345g.yaml:1]: Nit, also 2021

(Should have added the same comment here, sorry)

**Comment by galak** [comment]
Seeing a build failure on `samples/boards/stm32/power_mgmt/blinky/sample.boards.stm32.power_mgmt.blinky` which I've asked @erwango to look into.
